frontend/src/store/shoot/index.js

```diff
@@ -302,24 +299,45 @@ export const useShootStore = defineStore('shoot', () => {
       return { shoots: items, issues, comments: [] }
     }
 
+    const getThrottleDelay = (options, n) => {
+      if (options.name) {
+        return 0
+      }
+      const p = Math.pow(10, Math.floor(Math.log10(n)))
+      const d = 2 * p * Math.round(n / p)
+      return Math.min(5_000, Math.max(200, d))
+    }
+
     // await and handle response data in the background
     const fetchData = async options => {
+      let throttleDelay
       try {
-        setSubscriptionState(constants.LOADING)
+        setSubscriptionState(state, constants.LOADING)
         const promise = options.name
           ? fetchShoot(options)
-          : fetchShoots(options)
+          : fetchShoots({
+            useCache: localStorageStore.shootListFetchFromCache,
+            ...options,
+          })
         const { shoots, issues, comments } = await promise
-        receive(shoots)
+        shootStore.receive(shoots)
         ticketStore.receiveIssues(issues)
         ticketStore.receiveComments(comments)
-        openSubscription(options)
+        throttleDelay = getThrottleDelay(options, shoots.length)
       } catch (err) {
-        const message = get(err, 'response.data.message', err.message)
-        logger.error('Failed to fetch shoots or tickets: %s', message)
-        setSubscriptionError(err)
-        clearAll()
+        if (options.name && isNotFound(err)) {
+          throttleDelay = 0
+        } else {
+          const message = get(err, 'response.data.message', err.message)
+          logger.error('Failed to fetch shoots or tickets: %s', message)
+          setSubscriptionError(state, err)
+          shootStore.clearAll()
+        }
         throw err
+      } finally {
+        if (typeof throttleDelay === 'number') {
+          await shootStore.openSubscription(options, throttleDelay)
```

Comment:
I would like to pospone the feature of having a dynamic throttle delay. The lodash throttle implementation does not allow to control the delay e.g via a callback. This would mean to cancel running throttle timeouts and making sure that nothing is lost or processed twice.

Reply:
Opened issue https://github.com/gardener/dashboard/issues/1654